client/src/components/agent_components/AgentPatientSearchBar.tsx/AgentPatientSearchBar.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState, useCallback } from 'react';
+import { useGetPatientsByNameLazyQuery } from '../../../generated/graphql-types';
+import AgentSearchBar from '../AgentSearchBar.tsx/AgentSearchBar';
+import AgentPatientSearchBarProps from './AgentPatientSearchBar.type';
+
+export default function AgentPatientSearchBar({
+  handlePatientSelected
+}: AgentPatientSearchBarProps) {
+  const [search, setSearch] = useState<string>('');
+  const [getPatientsByName, { data }] = useGetPatientsByNameLazyQuery();
+
+  const formatSSN = (value: string): string => {
+    const cleanedValue = value.replace(/\s+/g, '');
+    let formattedValue = '';
+
+    for (let i = 0; i < cleanedValue.length; i++) {
+      formattedValue += cleanedValue[i];
+      if ([0, 2, 4, 6, 9, 12].includes(i)) {
+        formattedValue += ' ';
+      }
+    }
+
+    return formattedValue.trim();
+  };
+
+  const handleChange = (value: string): void => {
+    const sanitisedValue = formatSSN(value.replace(/\D/g, ''));
+    setSearch(sanitisedValue);
+  };
+
+  const handleSearch = useCallback((): void => {
```

Comment:
Probably no need for useCallback, move handleSearch logic directly into the useEffect line 38